### PR TITLE
.Net: Combining memory connector examples into TextMemorySkill 

### PR DESCRIPTION
**Pull request by shawncal**
Combining all the (nearly identical) memory connector syntax examples into the single Example15_TextMemorySkill.cs example.

### Motivation and Context
General cleanup and code reduction. 

### Contribution Checklist

- [X] The code builds clean without any errors or warnings
- [X] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [X] All unit tests pass, and I have added new tests where possible
- [X] I didn't break anyone :smile:

### CODE REVIEW

File: dotnet/samples/KernelSyntaxExamples/Example15_TextMemorySkill.cs [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+    }
+
+    private static async Task<IMemoryStore> CreateSamplePostgresMemoryStoreAsync()
+    {
+        NpgsqlDataSourceBuilder dataSourceBuilder = new(TestConfiguration.Postgres.ConnectionString);
+        dataSourceBuilder.UseVector();
+        await using NpgsqlDataSource dataSource = dataSourceBuilder.Build();
+        IMemoryStore store = new PostgresMemoryStore(dataSource, vectorSize: 1536, schema: "public");
+        return store;
+    }
```
Comment (dmytrostruk) [inline dotnet/samples/KernelSyntaxExamples/Example15_TextMemorySkill.cs:132]: Redis and Postgres connectors will throw an exception on a first request, because here we are disposing `ConnectionMultiplexer` in case of Redis and `NpgsqlDataSource` in case of Postgres at the end of these initialization methods. We should initialize them on higher level (maybe at the beginning of `RunAsync` method) and use them here or skip disposal since these are just samples.

File: dotnet/samples/KernelSyntaxExamples/Example15_TextMemorySkill.cs [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+    private static IMemoryStore CreateSampleKustoMemoryStore()
+    {
+        var connectionString = new Kusto.Data.KustoConnectionStringBuilder(TestConfiguration.Kusto.ConnectionString).WithAadUserPromptAuthentication();
+        IMemoryStore store = new KustoMemoryStore(connectionString, "MyDatabase");
+        return store;
+    }
+
+    private static async Task RunWithStoreAsync(IMemoryStore memoryStore, CancellationToken cancellationToken)
+    {
+        var logger = ConsoleLogger.Logger;
```
Comment (dmytrostruk) [inline dotnet/samples/KernelSyntaxExamples/Example15_TextMemorySkill.cs:143]: I just tried to run example with Chroma and this logger outputs too many messages with transient errors and warnings, so console window is overloaded and it's hard to find actual function results. What if we set `NullLogger.Instance` by default and provide a comment that for detailed logs it's possible to use `ConsoleLogger.Logger`? 

It applies for loggers on Kernel and each memory store level.

**Review by dmytrostruk — CHANGES_REQUESTED** [review]
I did a manual testing with Chroma and it works only partially :) 
For some reason during last requests, it tries to search memory record in collection with name `generics` which doesn't exist. 

Previous examples were simpler - save records, get records, search, output collections.
This example includes semantic function invocation and responses are not that reliable. Most probably we will have to re-test each memory connector for this scenario and check how each works. Otherwise, we may receive new issues that samples don't work as expected.

**Comment by shawncal** [comment]
> 

I'll try pulling in the examples that the other memory providers were following. Those should be simpler.

Thanks for running!

Flipping to Draft for now

File: dotnet/samples/KernelSyntaxExamples/Example15_TextMemorySkill.cs (hunk 2 above)
Comment (lemillermicrosoft) [reply dotnet/samples/KernelSyntaxExamples/Example15_TextMemorySkill.cs:143]: Aside: I've been updating ConsoleLogger locally to expose the LogFactory so I can create loggers more easily and tune filters per category/name/type. I'll try and get that pulled out of a draft and merged with the recent ILogger changes.

File: samples/notebooks/dotnet/config/settings.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "openai",
```
Comment (lemillermicrosoft) [inline samples/notebooks/dotnet/config/settings.json:2]: Is this related? or included on accident?

**Review by lemillermicrosoft — APPROVED** [review]

**Review by dmytrostruk — APPROVED** [review]
Tested manually with Redis, Chroma and Postgres.

**Review by hario90 — APPROVED** [review]

**Review by RogerBarreto — APPROVED** [review]
Not the scope of this PR but I'm not a very fan of leaving commented code in the examples for the other alternative examples. 

Since this is a console app, we could introduce some command interactions allowing the user to chose which memory it want to run.
